test/integration/clientset_test.go

```diff
@@ -47,506 +48,569 @@ type ipStruct struct {
 	Values map[string]string `json:"values"`
 }
 
-var instanceParameter = `{
+const (
+	instanceParameter = `{
     "bar": "barvalue",
     "values": {
       "first" : "firstvalue",
       "second" : "secondvalue"
     }
   }
 `
-
-// Used for testing binding parameters
-type bpStruct struct {
-	Foo string   `json:"foo"`
-	Baz []string `json:"baz"`
-}
-
-var bindingParameter = `{
+	bindingParameter = `{
     "foo": "bar",
     "baz": [
       "first",
       "second"
     ]
   }
 `
+)
+
+var storageTypes = []server.StorageType{
+	server.StorageTypeEtcd,
+	// TODO: enable this storage type. https://github.com/kubernetes-incubator/service-catalog/issues/407
+	// server.StorageTypeTPR,
+}
+
+// Used for testing binding parameters
+type bpStruct struct {
+	Foo string   `json:"foo"`
+	Baz []string `json:"baz"`
+}
 
 // TestGroupVersion is trivial.
 func TestGroupVersion(t *testing.T) {
-	client, shutdownServer := getFreshApiserverAndClient(t)
-	defer shutdownServer()
-
-	gv := client.Servicecatalog().RESTClient().APIVersion()
-	if gv.Group != servicecatalog.GroupName {
-		t.Fatal("we should be testing the servicecatalog group, not ", gv.Group)
+	rootTestFunc := func(sType server.StorageType) func(t *testing.T) {
+		return func(t *testing.T) {
+			client, shutdownServer := getFreshApiserverAndClient(t, sType.String())
+			defer shutdownServer()
+
+			gv := client.Servicecatalog().RESTClient().APIVersion()
```

Comment:
I would prefer to keep the programming model free of indirection for the actual test cases.  How about this: inject the `t` and `client` into a method called `do<name of test` - in this case `doTestGroupVersion`.  That keeps things a little easier to read and hides the complexity of multiple storage types if you're just looking at the test logic.